pytracking/tracker/dimp/dimp.py

```diff
@@ -252,60 +251,62 @@ def localize_advanced(self, scores, sample_pos, sample_scales):
-            return translation_vec1, scale_ind, scores_hn, 'not_found'
+            return translation_vec1, scale_ind, scores_hn, 'not_found', max_score1.item()
         if max_score1.item() < self.params.get('uncertain_threshold', -float('inf')):
-            return translation_vec1, scale_ind, scores_hn, 'uncertain'
+            return translation_vec1, scale_ind, scores_hn, 'uncertain', max_score1.item()
         if max_score1.item() < self.params.get('hard_sample_threshold', -float('inf')):
-            return translation_vec1, scale_ind, scores_hn, 'hard_negative'
+            return translation_vec1, scale_ind, scores_hn, 'hard_negative', max_score1.item()
 
         # Mask out target neighborhood
-        target_neigh_sz = self.params.target_neighborhood_scale * (self.target_sz / sample_scale) * (output_sz / self.img_support_sz)
+        target_neigh_sz = self.params.target_neighborhood_scale * (self.target_sz / sample_scale) * (
+                    output_sz / self.img_support_sz)
 
         tneigh_top = max(round(max_disp1[0].item() - target_neigh_sz[0].item() / 2), 0)
         tneigh_bottom = min(round(max_disp1[0].item() + target_neigh_sz[0].item() / 2 + 1), sz[0])
         tneigh_left = max(round(max_disp1[1].item() - target_neigh_sz[1].item() / 2), 0)
         tneigh_right = min(round(max_disp1[1].item() + target_neigh_sz[1].item() / 2 + 1), sz[1])
         scores_masked = scores_hn[scale_ind:scale_ind + 1, ...].clone()
-        scores_masked[...,tneigh_top:tneigh_bottom,tneigh_left:tneigh_right] = 0
+        scores_masked[..., tneigh_top:tneigh_bottom, tneigh_left:tneigh_right] = 0
 
         # Find new maximum
         max_score2, max_disp2 = dcf.max2d(scores_masked)
         max_disp2 = max_disp2.float().cpu().view(-1)
         target_disp2 = max_disp2 - score_center
         translation_vec2 = target_disp2 * (self.img_support_sz / output_sz) * sample_scale
 
-        prev_target_vec = (self.pos - sample_pos[scale_ind,:]) / ((self.img_support_sz / output_sz) * sample_scale)
+        prev_target_vec = (self.pos - sample_pos[scale_ind, :]) / ((self.img_support_sz / output_sz) * sample_scale)
 
         # Handle the different cases
         if max_score2 > self.params.distractor_threshold * max_score1:
-            disp_norm1 = torch.sqrt(torch.sum((target_disp1-prev_target_vec)**2))
-            disp_norm2 = torch.sqrt(torch.sum((target_disp2-prev_target_vec)**2))
+            disp_norm1 = torch.sqrt(torch.sum((target_disp1 - prev_target_vec) ** 2))
+            disp_norm2 = torch.sqrt(torch.sum((target_disp2 - prev_target_vec) ** 2))
             disp_threshold = self.params.dispalcement_scale * math.sqrt(sz[0] * sz[1]) / 2
 
             if disp_norm2 > disp_threshold and disp_norm1 < disp_threshold:
-                return translation_vec1, scale_ind, scores_hn, 'hard_negative'
+                return translation_vec1, scale_ind, scores_hn, 'hard_negative', max_score1.item()
```

Comment:
Lets talk about this section (where you return max score 1 and where max score 2)